src/hooks/useWidth.js

```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect } from "react";
+
+export default function useWidth() {
+  const [width, setWidth] = useState({ width: 0 });
+
+  useEffect(() => {
+    function handleWidthResize() {
+      setWidth({ width: window.innerWidth });
+    }
+    window.addEventListener("resize", handleWidthResize);
```

Comment:
May be this is not the right approach for using event listener

Reply:
But there is only this way  to check the window.width and that's a side-effect so i use Use-effect here .